Source/Core/AbsyQuant.cs

```diff
@@ -575,8 +575,15 @@ public Trigger(IToken/*!*/ tok, bool pos, IEnumerable<Expr>/*!*/ tr, Trigger nex
       }
     }
 
+    // todo (MR) this was implemented to throw an error; is this bad?
```

Comment:
Your new code is Good. It appears that it was not needed before, because then there had been no need for a map from "holes" to expressions. (A possible alternative, but I'm not pressing for it, might have been to have iterated through the triggers (via their `.Next` field) and to have visited their expressions lists `.tr`. That might have avoided the need for putting the triggers themselves into the hashtable. But the way you changed it is fine.)